src/Model/EditorGroup.re

```diff
@@ -82,6 +82,38 @@ let _getAdjacentEditor = (editor: int, reverseTabOrder: list(int)) => {
   };
 };
 
+let setActiveEditorByIndexDiff = (diff, model) => {
+  let tabs = model.reverseTabOrder;
+  let count = List.length(tabs);
+
+  if (count <= 1) {
+    // Nothing to change
+    model
+  } else {
+    switch (model.activeEditorId) {
+    | Some(activeEditorId) =>
+      switch (_getIndexOfElement(activeEditorId, tabs)) {
+      | (-1) => model
+      | idx =>
```

Comment:
Is there a good reason not to just resolve the TODO and use `List.find_opt`?

Reply:
Overall since this was my first submission and I am still learning ReasonML, I was trying to avoid too much refactoring to add to the noise of the diff.

Reply:
Yeah, I'm definitely not expecting you to do that. But if you want to, I'd encourage you to go for it.

> Is there a good reason not to just resolve the TODO and use List.find_opt?

Hmm, not that I can see. Apparently I'm the one who added that comment as well, but beats me why I did that instead of just changing it right away. Perhaps I wanted to avoid adding noise to the diff too. But it ought to be done at some point, so now seems as good a time as any.

Reply:
It turns out `List.find_opt` is not quite what is needed. Given an editor id, it just returns Some(id), or None, which is pretty useless here since we want the index. In fact I put it into use without thinking about it, with some hilarious and confusing results. It happens to pass type checking since both the editor ID and the index are `int`.

I think I will just change `_getIndexOfElement` to return `option(int)`. I'm not sure why that wasn't done originally...

Reply:
Ah, yeah good call! Dealing with indexes isn't a very natural use for lists, which is why something like this doesn't exist already in the standard library. It might be that the overall model here is a bit off, but just converting this to `option(int)` is a good way to address the immediate concern I think. :+1:

Reply:
I agree we may want to adjust the model data structure given this and the later need for setting the active editor by index. I am not yet sure what that model should be. But as you say moving to `option(int)` improves things for now.